tools/toolchains/gcc.py

```diff
@@ -177,7 +194,10 @@ def get_compile_options(self, defines, includes, for_asm=False):
     @hook_tool
     def assemble(self, source, object, includes):
         # Build assemble command
-        cmd = self.asm + self.get_compile_options(self.get_symbols(True), includes) + ["-o", object, source]
+        if self.check_if_coverage_enabled(source):
+            cmd = self.asm + self.COVERAGE_FLAGS + self.get_compile_options(self.get_symbols(True), includes) + ["-o", object, source]
+        else:
+            cmd = self.asm + self.get_compile_options(self.get_symbols(True), includes) + ["-o", object, source]
```

Comment:
Nit: When it comes to compiler flags, we probably want to avoid copy and pasting as much as possible. I'd suggest the following (though this isn't an absolutely necessary change, shouldn't block this PR):

```
cmd = self.asm

if self.check_if_coverage_enabled(source):
    cmd += self.COVERAGE_FLAGS

cmd += self.get_compile_options(self.get_symbols(True), includes) + ["-o", object, source]
```